src/impl/device/motor/motorGroup.cpp

```diff
@@ -9,6 +9,21 @@
 
 namespace okapi {
 MotorGroup::MotorGroup(const std::initializer_list<Motor> &imotors,
+                       const std::shared_ptr<Logger> &logger) {
+  if (imotors.size() == 0) {
+    std::string msg(
+      "MotorGroup: A MotorGroup must be created with at least one motor. No motors were given.");
+    LOG_ERROR(msg);
+    throw std::invalid_argument(msg);
+  }
+
+  motors.reserve(imotors.size());
+  for (auto motor : imotors) {
+    motors.push_back(std::make_shared<Motor>(motor));
```

Comment:
I could be mistaken, but I believe in the `MotorGroup` ctor, the copy constructor of the input motors is called twice. 
It gets first copied when the motors get passed through the `initializer_list`, it then gets copied again when `make_shared` allocates new memory. I might be being silly and this is negligible, but I think it might be a good idea to pass the motors through the initializer list by universal reference.

Reply:
I think it's just once when the shared_ptr is made. We need to make at least one copy, so I am not that concerned.